plugins/modules/cloud/lxd/lxd_container.py

```diff
@@ -356,8 +372,8 @@ def __init__(self, module):
         self.addresses = None
         self.target = self.module.params['target']
 
-        self.key_file = self.module.params.get('client_key', None)
-        self.cert_file = self.module.params.get('client_cert', None)
+        self.key_file = self.module.params.get('client_key', '{0}/.config/lxc/client.key'.format(os.environ['HOME']))
```

Comment:
That won't work, since `client_key` is always present in the dictionary (with value `None`). You need to do this manually:
```suggestion
        self.key_file = self.module.params.get('client_key')
        if self.key_file is None:
            self.key_file = '{0}/.config/lxc/client.key'.format(os.environ['HOME']))
```
Same for the next line.

Reply:
done this